### PR TITLE
:sparkles: Udp init refactor

### PR DESCRIPTION
**Pull request by Tugduoff**

### CODE REVIEW

File: src/client/Main.cpp [hunk 1]
```diff
@@ -35,38 +35,39 @@ int main()
...
-    engine.registerComponent<Components::Position>("./plugins/bin/components/", "Position");
-    engine.registerComponent<Components::SpriteID>("./plugins/bin/components/", "SpriteID");
-    engine.registerComponent<Components::DeathRange>("./plugins/bin/components/", "DeathRange");
-    engine.registerComponent<Components::Sound>("./plugins/bin/components/", "Sound");
-    engine.registerComponent<Components::Destruction>("./plugins/bin/components/", "Destruction");
 
-    conn.engineInit();
-    std::unordered_map<uint8_t, std::string> compNames =  conn.getCompNames();
+    conn.asyncReceive(engine);
+    // conn.engineInit(engine.getIdStringToType());
```
Comment (Tholarr) [inline src/client/Main.cpp:40]: comment left on purpose ?

File: src/client/Main.cpp [hunk 2]
```diff
@@ -35,38 +35,39 @@ int main()
...
+    conn.send(std::vector<uint8_t>({0x2}));
+    conn._compNamesByNetwork.clear();
+
     try
     {
         // Check that you have the same components here with the map in RTypeClient
         while (conn.gameEnd != true) {
-            if (chrono.getElapsedTime() < 17)
-                continue;
+            // std::cout << "Packet queue size: " << conn._packetQueue.size() << std::endl;
```
Comment (Tholarr) [inline src/client/Main.cpp:66]: comment left on purpose ?

File: src/network/RTypeClient.hpp [hunk 3]
```diff
@@ -35,11 +36,16 @@ class RTypeClient : public UDPConnection
          */
         RTypeClient(std::string hostname, std::string port);
 
-        /**
-         * @brief This function gets all the types that will be used during program execution
-         * @brief and stores them in _compNames
-         */
-        void engineInit();
+        // /**
```
Comment (Tholarr) [inline src/network/RTypeClient.hpp:39]: comments left on purpose ? (line 39 to 48)

File: src/network/RTypeClient.hpp [hunk 4]
```diff
@@ -191,14 +199,22 @@ class RTypeClient : public UDPConnection
...
     public:
-        bool gameEnd;
-    
+        std::atomic_bool gameEnd;
+        std::atomic_bool nextFrame;
+        std::atomic_bool finishedInit;
+
+        std::vector<std::pair<std::string, uint16_t>> _compNamesByNetwork;
+        // std::queue<std::vector<uint8_t>> _packetQueue;
+        // std::mutex _packetQueueMutex;
```
Comment (Tholarr) [inline src/network/RTypeClient.hpp:208]: comments left on purpose ?

File: src/server/Main.cpp [hunk 5]
```diff
@@ -139,10 +147,13 @@ int main() {
         server.updateIdStringToType(engine.getIdStringToType());
 
         // engine.getRegistry().componentManager().
```
Comment (Tholarr) [inline src/server/Main.cpp:149]: comment left on purpose ?

**Review by ClementBaziret — APPROVED** [review]
Seems good to me :saluting_face:

**Review by tibaudlemaout — APPROVED** [review]

File: src/client/Main.cpp (hunk 1 above)
Comment (ClementBaziret) [reply src/client/Main.cpp:90]: yes

Comment (ClementBaziret) [reply src/client/Main.cpp:90]: yes

File: src/client/Main.cpp (hunk 2 above)
Comment (ClementBaziret) [reply src/client/Main.cpp:116]: tyes

File: src/network/RTypeClient.hpp (hunk 3 above)
Comment (ClementBaziret) [reply src/network/RTypeClient.hpp:41]: yes

File: src/network/RTypeClient.hpp (hunk 4 above)
Comment (ClementBaziret) [reply src/network/RTypeClient.hpp:214]: yes

File: src/server/Main.cpp (hunk 5 above)
Comment (ClementBaziret) [reply src/server/Main.cpp:149]: yes

**Review by ClementBaziret — APPROVED** [review]
Seems good to me :saluting_face:

**Review by C0Florent — APPROVED** [review]

**Pull request merged**